### PR TITLE
Impl server.properties dec/encoder.

### PR DESCRIPTION
**Pull request by toqueteos**
As seen in #40.

Needed for #44.

### CODE REVIEW

File: server.properties [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+resource-pack=
+spawn-animals=true
+white-list=false
+generate-structures=true
+online-mode=false
+max-build-height=256
+level-seed=1122334455667788
+use-native-transport=true
+motd=A Minecraft Server made in Rust!
+enable-rcon=false
```
Comment (fenhl) [inline server.properties]: What is the purpose of this file?

Comment (toqueteos) [inline server.properties]: Mmmm maybe I should put it inside `tests/`. We do need one for SLP testing.

Comment (fenhl) [inline server.properties]: How about a directory for an example server? The favicon should be moved there too.

Comment (toqueteos) [inline server.properties]: The nice thing about having both `server.properties` and `favicon.png` on root is that we can do `cargo run` and it just works.

We could drop the `server.properties` file but keep favicon maybe with the project's logo (when we have one).

**Comment by toqueteos** [comment]
Added newlines to encoder and got rid of `server.properties` file.
`server/main.rs` will have to deal with it existing or not.

**Comment by eddyb** [comment]
That's... a lot of manual decoding. I thought I've shown you [this macro-based Encodable & Decodable implementation](https://gist.github.com/eddyb/6c6acd7e6f5738dbd333#file-xtb-rs-L133).

**Comment by toqueteos** [comment]
@eddyb One line of code per property, macro's wouldn't help that much here, I think. It's mostly calls to parse.

**Comment by toqueteos** [comment]
Woops, forgot the `.unwrap` calls! `try!`ing all of those.

**Comment by eddyb** [comment]
I see 4 lines per property, not counting tests. I wouldn't trust myself to [get that repeating madness right](https://medium.com/@Code_Analysis/the-last-line-effect-7b1cb7f0d2a1).

**Comment by atheriel** [comment]
Does the `server.properties` file respect a traditional file format (TOML, INI, etc)? If so, there's probably a library somewhere that will do this for us...

**Comment by toqueteos** [comment]
@atheriel It seems it's a mostly Java thing http://en.wikipedia.org/wiki/.properties

**Comment by toqueteos** [comment]
Current macro-based proposal, as heavily suggested by @eddyb to avoid repetition: https://gist.github.com/toqueteos/a323e9bb1040a0f71f25

~~Line 52 is problematic.~~ Solved! Pushing...

**Comment by toqueteos** [comment]
Adding `$(#[$attr:meta])*` allows us to have comments for each struct field.

**Pull request merged**